AdvancedC#/FileSystemVisitorTests/FileSystemVisitorTests.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Xunit;
+
+namespace FileSystemVisitorTests;
+
+public class FileSystemVisitorTests
+{
+    private readonly string _testDirectory;
+
+    public FileSystemVisitorTests()
+    {
+        var tempPath = Path.GetTempPath();
+        _testDirectory = Path.Combine(tempPath, Guid.NewGuid().ToString());
+
+        Directory.CreateDirectory(_testDirectory);
```

Comment:
Honestly, Unit tests should never rely on real environment including system (the reason is restricted file access for app or user).
Instead of that, the easiest way is to create a **wrapper**: interface that defines contract you wish to have for search and it's implementation that encapsulates `Path` file. You can call it like `IFileExplorer`. This `IFileExplorer` should be injected directly in `FileSystemVisitor`.
The key idea of wrapper is to split real business logic from some static or native tools. To mock data you can use `Moq` or `FakeItEasy` (last one is my preference), it also simplifies assertion of your calls and making some fake.
Let's, please, try to add some simple file wrapper and use mocks for testing. The rest part looks great!